app/Models/Course.php

```diff
@@ -33,4 +38,98 @@ public function tags()
     {
         return $this->hasMany(Tag::class);
     }
+
+    public function lessons()
+    {
+        return $this->hasMany(Lesson::class);
+    }
+
+    public function scopeKeyWord($query, $keyword)
+    {
+        return $query->where('name', 'like', '%' . $keyword . '%')->orWhere('description', 'like', '%' . $keyword . '%');
+    }
+
+    public function scopeTeacher($query, $teacher)
+    {
+        if (isset($teacher)) {
+            return $query->where('teacher_id', $teacher);
+        }
+        return $query;
+    }
+
+    public function scopeTag($query, $tag)
+    {
+        if (isset($tag)) {
+            return $query->whereHas('tags', function ($q) use ($tag) {
+                $q->where('name', $tag);
+            });
+        }
+        return $query;
+    }
+
+    public function scopeCreatedAt($query, $createdAt)
+    {
+        if (isset($createdAt)) {
+            return $query->orderBy('created_at', $createdAt);
+        }
+        return $query;
+    }
+
+    public function scopeLearners($query, $learners)
+    {
+        if (isset($learners)) {
+            return $query->orderBy('users_count', $learners);
```

Comment:
mấy phần đếm tổng số người học, số giờ học vẫn đang dùng trường mặc định của db mà không tính toán ra này